app/models/host.rb

```diff
@@ -41,5 +39,22 @@ def filter_by_compliance_score(_filter, operator, score)
 
       { conditions: "hosts.id IN(#{ids.join(',')})" }
     end
+
+    def has_test_results(_filter, _operator, value)
+      operator = ActiveModel::Type::Boolean.new.cast(value) ? '' : 'NOT'
+      host_ids = TestResult.select(:host_id).distinct.where.not(:host_id => nil)
+      {
+        conditions: "hosts.id #{operator} "\
+                    "IN(#{host_ids.to_sql})"
```

Comment:
This generates a SQL subquery, so the whole thing is executed at once. I guess it's a valid workaround since scoped_search will just take this and `Host.where(conditions)`, so we can't really pass AR relations around here

Reply:
Yep, `to_sql` is a valid way to do this :+1: